docs/tutorial/index.rst

```diff
@@ -36,6 +36,18 @@ omitted in the case of trailing inputs:
 
 .. literalinclude:: examples/omitted_input.py
 
+**Using optional inputs**
+
+The example below shows a definition of ``Clip`` as an *onnxscript* function.
+The ``Clip`` operator has input arguments that are *optional*: the *min* and *max* inputs.
+``OptionalHasElement`` and ``OptionalGetElement`` operators can be used to utilize optional
+inputs within an *onnxscript* function, as shown below:
+
+.. literalinclude:: examples/clip.py
+
+This pattern will not work in eager-mode execution as
+``OptionalHasElement`` and ``OptionalGetElement`` operators do not take None as inputs.
```

Comment:
In opset 18 they do. But there is likely a different issue, which is to do with the two kinds of optional inputs we have in ONNX. Optional inputs of the first kind are supported only by ops/nodes, not by graphs. we may need some extra code in onnxscript when we turn a single-op invocation into a graph/model

Reply:
It may be simpler to provide a python implementation in the ORTMixedEvaluator, like [here](https://github.com/microsoft/onnx-script/blob/main/onnxscript/evaluator.py#L450) and use it

Reply:
I took a look at the onnxscript code, and it does seem to do what's necessary for optional inputs (so ignore my comment above). It may be worth trying using opset18 and see if it works.